src/Future.re

```diff
@@ -46,7 +46,8 @@ let make = (~executor: executorType=`none, resolver) => {
       callbacks := [];
     | Some(_) => () /* Do nothing; theoretically not possible */
     }
-  );
+  )
+  ->ignore;
```

Comment:
This is the main fix.  Without this BuckleScript 8 now doesn't resolve the type as a unit.